woocommerce/woocommerce-android
# Update/add login related mocks.

### Background 
Automating a selected range of UI tests is the project I'm currently part of. There was a foundation in place for the UI tests already, in the form of a number of screen objects, unfinished ScreenshotTest and numerous WireMock files.

First thing I tried is executing the ScreenshotTest (which currently only logs in). It turned out that both mocks and login sequence on UI level are outdated, so even the login was not working.

### Description
I'll provide more detailed comments on the code, but here's the short summary:

1. Used WireMock version has been bumped by Dependabot some time ago with #4025. However, it turns out that this update made the use of WireMock impossible (see wiremock/wiremock/issues/1345 for more). The used version is revered back in current PR.
2. The login flow in the app UI seems to no longer show the magic link screen after entering user email, and now goes directly to the password screen. This PR updates this.
3. A number of mocks were outdated and required a change in order to have the login mocked. Also, one new file was added.

I tried making the changes to the mocks minimal, but enough to have the login 'working'.

### Testing instructions
1. Use the `gradle.properties` from secrets
2. Run `ScreenshotTest` ~> It should now get past the login and reach the `My Store` screen. 

Please note that the test will still be marked as failed.

### Release
These changes do not require release notes.

### P.S.
Since this is my first PR in this repo, I might be doing some things wrong. I tried to follow the contribution guidelines, but let me know if there's a thing I should adjust. Also, this is my first experience with mocks - so I'll be happy to hear any constructive feedback on that too.
